app/javascript/src/AccessTokens/components/ExpirationDatePicker.tsx

```diff
@@ -32,46 +32,51 @@ interface Props {
 }
 
 const ExpirationDatePicker: FunctionComponent<Props> = ({ id, label, tzOffset }) => {
-  const [selectedItem, setSelectedItem] = useState(collection[0])
-  const [pickedDate, setPickedDate] = useState(new Date())
+  const [dropdownSelectedItem, setDropdownSelectedItem] = useState(collection[0])
+  const [calendarPickedDate, setCalendarPickedDate] = useState(new Date())
   const fieldName = `human_${id}`
   const fieldLabel = label ?? 'Expires in'
 
-  const fieldDate = useMemo(() => {
-    if (selectedItem.period === 0) return null
+  const dropdownDate = useMemo(() => {
+    if (dropdownSelectedItem.period === 0) return null
 
-    return new Date(new Date().getTime() + selectedItem.period * dayMs)
-  }, [selectedItem])
+    return new Date(new Date().getTime() + dropdownSelectedItem.period * dayMs)
+  }, [dropdownSelectedItem])
 
-  const dateValue = useMemo(() => {
-    let value = ''
+  const selectedDate= useMemo(() => {
+    let value = null
 
-    if (fieldDate) {
-      value = fieldDate.toISOString()
-    } else if (selectedItem.id === 'custom' ) {
-      value = pickedDate.toISOString()
+    if (dropdownDate) {
+      value = dropdownDate
```

Comment:
I didn't like the fact that we first convert the date to string, and then in `formattedDateValue` we pparsed it again into Date.
The idea is that "the source of truth" for the selected date is a `Date` instance, and then it is formatted for the representation purposes as the last step.